src/fides/api/util/tcf/tc_string.py

```diff
@@ -180,3 +188,87 @@ def build_disclosed_vendors_string(model: TCModel) -> str:
     return base64.urlsafe_b64encode(
         _convert_bitstring_to_bytes(disclosed_vendor_bits)
     ).decode()
+
+
+def boolean_to_user_consent_preference(preference: bool) -> UserConsentPreference:
+    return UserConsentPreference.opt_in if preference else UserConsentPreference.opt_out
+
+
+FidesPreferenceType = Union[TCFPurposeSave, TCFVendorSave, TCFSpecialFeatureSave]
+
+
+def convert_to_fides_preference(
+    datamap_options: List[int],
+    tc_string_preferences: Dict[int, bool],
+    preference_class: Union[
+        Type[TCFPurposeSave], Type[TCFVendorSave], Type[TCFSpecialFeatureSave]
+    ],
+) -> List[FidesPreferenceType]:
+    """For a field in the TC string, convert to a list of preferences where we opt-in if it is included in the string,
+    and opt-out if it is not in the string, but in the datamap"""
+    # Loop through all the datamap options that are currently available.  For example, all of the vendors
+    # from the datamap against which we can save preferences with a consent legal basis
+    preferences_array: List[FidesPreferenceType] = []
+
+    for identifier in datamap_options:
```

Comment:
This is important.  The TC string only encodes opt in preferences:  say this is my vendor preferences string: "001" for example.  All I know for sure from the string is that the user opted into vendor 3.  Vendors 1, and 2 may have been opted out, or they _might not have been presented to the user at all._

So I let the datamap drive what gets saved from the string.  The TCF Experience is built, then that section of the experience is compared to the string, so I know what options were disclosed to the user in the first place.  If Vendor 1 is in the Vendor Consent section of the experience, then I save an opt-out here.  If it wasn't in the experience at all, I save nothing.